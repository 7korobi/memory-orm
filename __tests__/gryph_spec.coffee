require "./models/index"
Mem = require "../src/index"

describe "gryph", ->
  test 'points snapshot', ->
    expect Mem.Query.points.reduce
    .toMatchSnapshot()
    return

  test 'point structure snapshot', ->
    expect Mem.Query.points.pluck "id", "x", "y"
    .toMatchSnapshot()
    return

  test 'lines snapshot', ->
    expect Mem.Query.lines.reduce
    .toMatchSnapshot()
    return

  test 'deltas snapshot', ->
    expect Mem.Query.deltas.reduce
    .toMatchSnapshot()
    return

  return