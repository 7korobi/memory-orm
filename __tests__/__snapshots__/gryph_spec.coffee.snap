// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gryph deltas snapshot 1`] = `
Object {
  "hash": Object {},
  "list": list [],
  "set": Array [],
}
`;

exports[`gryph lines snapshot 1`] = `
Object {
  "hash": Object {
    "1,-1*-1,1": model [
      Array [
        1,
        -1,
      ],
      Array [
        -1,
        1,
      ],
    ],
    "1,1*-1,-1": model [
      Array [
        1,
        1,
      ],
      Array [
        -1,
        -1,
      ],
    ],
    "1,1*1,-1*-1,1*-1,-1": model [
      Array [
        1,
        1,
      ],
      Array [
        1,
        -1,
      ],
      Array [
        -1,
        1,
      ],
      Array [
        -1,
        -1,
      ],
    ],
  },
  "list": list [
    model [
      Array [
        1,
        1,
      ],
      Array [
        -1,
        -1,
      ],
    ],
    model [
      Array [
        1,
        -1,
      ],
      Array [
        -1,
        1,
      ],
    ],
    model [
      Array [
        1,
        1,
      ],
      Array [
        1,
        -1,
      ],
      Array [
        -1,
        1,
      ],
      Array [
        -1,
        -1,
      ],
    ],
  ],
  "set": Array [
    "1,1*-1,-1",
    "1,-1*-1,1",
    "1,1*1,-1*-1,1*-1,-1",
  ],
}
`;

exports[`gryph point structure snapshot 1`] = `
list [
  Array [
    "1*1",
    1,
    1,
  ],
  Array [
    "1*-1",
    1,
    -1,
  ],
  Array [
    "-1*1",
    -1,
    1,
  ],
  Array [
    "-1*-1",
    -1,
    -1,
  ],
]
`;

exports[`gryph points snapshot 1`] = `
Object {
  "hash": Object {
    "-1*-1": model [
      -1,
      -1,
    ],
    "-1*1": model [
      -1,
      1,
    ],
    "1*-1": model [
      1,
      -1,
    ],
    "1*1": model [
      1,
      1,
    ],
  },
  "list": list [
    model [
      1,
      1,
    ],
    model [
      1,
      -1,
    ],
    model [
      -1,
      1,
    ],
    model [
      -1,
      -1,
    ],
  ],
  "set": Array [
    "1*1",
    "1*-1",
    "-1*1",
    "-1*-1",
  ],
}
`;
